src/reducers/versions.tsx

```diff
@@ -242,9 +242,13 @@ export const actions = {
       resolve(payload);
   }),
   loadVersionInfo: createAction('LOAD_VERSION_INFO', (resolve) => {
-    return (payload: {
+    return ({
+      updatePathInfo = true,
```

Comment:
Thanks for the update @mirefly. I know I asked for the default to be more explicit, but I find this way of doing it to be inconsistent with the way we generally define actions, to the point that it is confusing.

I still feel that making it non-optional would be best. I wonder if there is a way that you can solve the problem of `loadVersionInfo` being used frequently in tests with a helper? If not, perhaps the default can be set in the `case` statement for `loadVersionInfo`, rather than in the action definition?

What do you think?

Reply:
>  I find this way of doing it to be inconsistent with the way we generally define actions, to the point that it is confusing.

I agree.

@bobsilverberg I come up with this approach, but not very sure. How do you think about it?
``` diff
diff --git a/src/reducers/versions.spec.tsx b/src/reducers/versions.spec.tsx
index 53d499d..7ad545f 100644
--- a/src/reducers/versions.spec.tsx
+++ b/src/reducers/versions.spec.tsx
@@ -13,7 +13,7 @@ import reducer, {
   Version,
   VersionEntryStatus,
   VersionEntryType,
-  actions,
+  _actions as actions,
   createEntryStatusMap,
   createInternalDiff,
   createInternalHunk,
diff --git a/src/reducers/versions.tsx b/src/reducers/versions.tsx
index 9978794..8c54da2 100644
--- a/src/reducers/versions.tsx
+++ b/src/reducers/versions.tsx
@@ -242,13 +242,10 @@ export const actions = {
       resolve(payload);
   }),
   loadVersionInfo: createAction('LOAD_VERSION_INFO', (resolve) => {
-    return ({
-      updatePathInfo = true,
-      ...moreParams
-    }: {
+    return (payload: {
       version: ExternalVersionWithContent | ExternalVersionWithDiff;
-      updatePathInfo?: boolean;
-    }) => resolve({ updatePathInfo, ...moreParams });
+      updatePathInfo: boolean;
+    }) => resolve(payload);
   }),
   loadEntryStatusMap: createAction('LOAD_ENTRY_STATUS_MAP', (resolve) => {
     return (payload: {
@@ -344,6 +341,17 @@ export const actions = {
   ),
 };
 
+export const _actions: typeof actions = {
+  ...actions,
+  loadVersionInfo: ({
+    updatePathInfo = true,
+    version,
+  }: {
+    updatePathInfo?: boolean;
+    version: ExternalVersionWithContent | ExternalVersionWithDiff;
+  }) => actions.loadVersionInfo({ updatePathInfo, version }),
+};
+
 export type VersionsState = {
   byAddonId: {
     [addonId: number]: VersionsMap;
@@ -1098,6 +1106,7 @@ export const fetchDiff = ({
         dispatch(
           actions.loadVersionInfo({
             version: response,
+            updatePathInfo: true,
           }),
         );
       }

```

Reply:
This is clever, but I fear it may be a bit too clever. It still differs greatly from anything else we have, and that can make it difficult to follow for anyone looking at the code for the first time.

What if you were to create a helper called `_loadVersionInfo` in the tests that calls `actions.loadVersionInfo` with a default of `true` for `updatePathInfo`? That might remove the burden from the tests and still allow us to define `updatePathInfo` as a non-optional argument in a standard way.

Reply:
I am not sure where to define `_loadVersionInfo`, since it will be used in a few test files, `test-helper.tsx` or `version.tsx` (or `version.spec.tsx`). But in either way, we have to import `_loadVersionInfo` as a special case, different from all other actions (I do not like it...). Let me think about it again.

Reply:
Yes, I see your point. Could you post a few links to places where we currently need to use `loadVersionInfo` in tests and I can see if I can come up with an idea of making a helper work that will feel good to both of us?

Reply:
Here are some places we called `loadVersionInfo` in tests.
https://github.com/mozilla/addons-code-manager/blob/master/src/components/FileTree/index.spec.tsx#L36
https://github.com/mozilla/addons-code-manager/blob/master/src/components/Navbar/index.spec.tsx#L317
https://github.com/mozilla/addons-code-manager/blob/master/src/components/VersionFileViewer/index.spec.tsx#L68
https://github.com/mozilla/addons-code-manager/blob/master/src/pages/Browse/index.spec.tsx#L96
https://github.com/mozilla/addons-code-manager/blob/master/src/pages/Compare/index.spec.tsx#L428
https://github.com/mozilla/addons-code-manager/blob/master/src/reducers/fileTree.spec.tsx#L92
https://github.com/mozilla/addons-code-manager/blob/master/src/reducers/versions.spec.tsx#L226
https://github.com/mozilla/addons-code-manager/blob/master/src/test-helpers.tsx#L873

Reply:
Thanks @mirefly. I'll take a look and see if I have any good ideas. In the meantime, what did you think about the idea of dealing with the default value in the `case` statement?

Reply:
> what did you think about the idea of dealing with the default value in the case statement?

That will be easiest way! I am Ok with that approach.
I just want to make sure you mean something like this, right?
```diff

diff --git a/src/reducers/versions.tsx b/src/reducers/versions.tsx
index acf5c2e..11b4957 100644
--- a/src/reducers/versions.tsx
+++ b/src/reducers/versions.tsx
@@ -245,6 +245,7 @@ export const actions = {
   loadVersionInfo: createAction('LOAD_VERSION_INFO', (resolve) => {
     return (payload: {
       version: ExternalVersionWithContent | ExternalVersionWithDiff;
+      updatePathInfo?: boolean;
     }) => resolve(payload);
   }),
   loadEntryStatusMap: createAction('LOAD_ENTRY_STATUS_MAP', (resolve) => {
@@ -1257,6 +1258,19 @@ export const createReducer = ({
       case getType(actions.loadVersionInfo): {
         const { version } = action.payload;
         const internalVersion = createInternalVersion(version);
+
+        let { updatePathInfo } = action.payload;
+        updatePathInfo = updatePathInfo ? updatePathInfo : true;
+        if (updatePathInfo) {
+          return {
+            ...state,
+            versionInfo: {
+              ...state.versionInfo,
+              [version.id]: internalVersion,
+            },
+          };
+        }
+
         const {
           expandedPaths: expandedPathsInState,
           selectedPath: selectedPathInState,


```

Reply:
@bobsilverberg I reconsidered all approaches we discussed so far. I personally like the idea of dealing with default value in `case` statement most.

I might have to add another optional parameter for `loadVersionInfo` in fixing #932 . It is unpleasant to update default values in several places.